android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -219,6 +210,113 @@ public void setUrlPrefixesForDefaultIntent(ReadableArray specialUrls) {
     }
   }
 
+  protected static class RNCWebChromeClient extends WebChromeClient {
+    protected static final FrameLayout.LayoutParams FULLSCREEN_LAYOUT_PARAMS = new FrameLayout.LayoutParams(
+            LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT, Gravity.CENTER);
+
+    protected ReactContext mReactContext;
+    protected RNCWebViewModule mModule;
+    protected View mWebView;
+
+    protected View mVideoView;
+    protected WebChromeClient.CustomViewCallback mCustomViewCallback;
+
+    public RNCWebChromeClient(ReactContext reactContext, RNCWebViewModule module, WebView webView) {
+      this.mReactContext = reactContext;
+      this.mModule = module;
+      this.mWebView = webView;
+    }
+
+    @Override
+    public void onShowCustomView(View view, CustomViewCallback callback) {
+      if (mVideoView != null) {
+        callback.onCustomViewHidden();
+        return;
+      }
+
+      mVideoView = view;
+      mCustomViewCallback = callback;
+
+      mReactContext.getCurrentActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
```

Comment:
Trying to force the activity to landscape orientation seems to cause an issue with some video players (i.e. YouTube) if the video was recorded in portrait orientation. In my testing the video starts to go fullscreen and then immediately closes as (I assume) the video player tries to change the orientation to portrait. Removing this line starts all videos at the orientation of the parent view and then allows the user to change the orientation of the device if they wish which seems to fix the issue.

Reply:
Interesting. I didn't test with portrait video - I'll give that a go. What I do know is that when I tested on an older device with a custom ROM (so take with a grain of salt), is that landscape videos on YouTube had graphical glitching when minimizing the video, unless a rotation happened at some point after. I figured most videos would be landscape anyway, and forcing the orientation seemed to fix it.

I will try to test tomorrow at work on more of our QA phones to see if I can get more information on what's going on. Maybe it would be reasonable to choose landscape if it's a landscape video and portrait if not? Or would you prefer that, if possible, no orientation is forced?

Reply:
I think it's reasonable to change the orientation based on the content of the video if we can do that. I'll bet the vast majority of users will re-orient the device for landscape videos anyway. It also looks like if you open a portrait video and then turn the phone to landscape the video closes on it's own which seems really strange.

Reply:
After some more research, I've found the cause of the video glitching after minimizing. I don't know why it happens, but if you don't hide the system UI/status bar, and minimize from fullscreen, it glitches hard (the video acts as though it is larger than the `<video>` element, making the controls impossible to reach unless you get through them by rotating). Was able to reproduce reliably across two different devices. Sticky immersive mode fixes this (and seems to be what Chrome uses), but we have to handle the user possibly switching apps or locking the screen and coming back to re-hide the system UI. That's only available in KitKat or newer, so I have to figure out how it works for pre-KitKat. In some cases in multi-window, it still glitches. I don't yet know how to fix that.

I didn't have much luck at being able to determine whether a video is portrait or landscape. Someone with more Android chops might need to do that. So no more forcing orientation.

I'll be making these changes within the next day or two hopefully.